server/services/applicationService.ts

```diff
@@ -75,9 +74,22 @@ export default class ApplicationService {
     }
   }
 
-  private saveToSession(page: TasklistPage, request: Request) {
-    request.session = this.fetchOrInitializeSessionData(request.session, request.params.task, request.params.id)
-    request.session.application[request.params.id][request.params.task][request.params.page] = page.body
+  private getApplicationFromSessionOrAPI(request: Request): Promise<Application> {
+    const { application } = request.session
+
+    if (application && application.id === request.params.id) {
+      return application as unknown as Promise<Application>
+    }
+    return this.findApplication(request.user.token, request.params.id)
```

Comment:
I think we should be awaiting this

Reply:
This gives me `Redundant use of 'await' on a return value`

Reply:
Oh yeah, I think you'd have to store is as a const first and then return the const (for reasons I don't understand). As long as we're awaiting the value returns from the function it will be fine